### PR TITLE
Update scine-qcmaquis

### PR DESCRIPTION
**Pull request by lorisercole**
Version 3.1.4 and 3.1.2 of `scine-qcmaquis` are added.
The option to build the OpenMolcas interface is added, and some dependencies are clarified.

### CODE REVIEW

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py:20]: Is the removal of this variant intentional?  Does it not still apply to at least the older versions?

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py:22]: Is it deliberate to no longer have a preferred version?

**Review by tldahlgren — COMMENTED** [review]
Confirmed the new branches exist.

I have some questions and, given the extent of the changes, will defer to @adam-grofe as the maintainer to do a proper review.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py
Comment (lorisercole) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py:20]: The `blas` variant was used to select the BLAS/LAPACK library to use. However, it does not define any input for Cmake, which detects this library automatically. QCMaquis does not specify a default/preferred one.
The user can decide whether to use OpenBlas or MKL by specifying `^openblas` or `^intel-oneapi-mkl` in the spec, or they can leave the choice to spack's concretizer. So I do not see the need for a `blas` variant.
Let me know what is Spack's suggested approach in this case.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py
Comment (lorisercole) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py:22]: I do not see why the old `3.1.2` version should be the preferred one.
I'll let @adam-grofe comment if there was a reason for this choice.

Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py:22]: If there isn't a reason to use the latest, then I don't see why the latest wouldn't be preferred.

Comment (lorisercole) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py:22]: Not specifying a preferred one means that the newest version will be chosen, as far as I understand.

Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py:22]: Yes, I confirmed in the docs. [here](https://spack.readthedocs.io/en/latest/packaging_guide.html#version-selection)

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
